### ISSUE TITLE
[MDEP-94] Add dependency:tree goal

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-dependency-plugin#637]
**[Mark Hobson](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=mihobson)** opened **[MDEP-94](https://issues.apache.org/jira/browse/MDEP-94?redirect=false)** and commented

The attached patch adds the help:dependencies goal from the maven-help-plugin to the maven-dependency-plugin as dependency:tree, as discussed on the dev mailing list.

---

**Affects:** 2.0

**Attachments:**
- [MDEP-94-test.patch](https://issues.apache.org/jira/secure/attachment/12714969/MDEP-94-test.patch) (_10.19 kB_)
- [patch.txt](https://issues.apache.org/jira/secure/attachment/12715116/patch.txt) (_10.26 kB_)

**Issue Links:**
- [MPH-24](https://issues.apache.org/jira/browse/MPH-24) Move help:dependencies goal to maven-dependency-plugin
 (_**"is depended upon by"**_)

1 votes, 1 watchers

**Comment by jira-importer** [comment]
**[Mark Hobson](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=mihobson)** commented

Attached patch for simple test in the same vein as the others.  Needed to up maven-plugin-testing-harness to 1.1-SNAPSHOT for a more complete MavenProject stub.  May be worth centralising all stubs in maven-plugin-testing-harness?

**Comment by jira-importer** [comment]
**[Brian E. Fox (imported)](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=brianf@infinity.nu)** commented

Yes, I need to update the unarchiver version, which brings in a new container that needs a new harness, so it's on my roadmap to some how deal with the unit tests in a more common way.

**Comment by jira-importer** [comment]
**[Mark Hobson](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=mihobson)** commented

I've just checked out a fresh trunk, applied the patch and successfully run the tests.  That exception is due to MavenProjectStub wrongly returning a singleton list of the empty string rather than the empty list, but this was fixed in MNG-2998.  Are you sure you've got the latest snapshot installed?
